apollo-router/tests/samples/basic/interface-object/plan.json

```diff
@@ -0,0 +1,254 @@
+                                        }
+                                    }
+                                },
+                                "response": {
+                                    "body": {
+                                        "data": {
+                                            "_entities": [
+                                                {
+                                                    "y": 1
+                                                }
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        ]
+                    },
+                    "reviews": {
+                        "requests": [
+                            {
+                                "request": {
+                                    "body": {
+                                        "query": "query TestItf2__reviews__2($representations:[_Any!]!){_entities(representations:$representations){...on C{c}}}",
+                                        "operationName": "TestItf2__reviews__2",
+                                        "variables": {
+                                            "representations": [
+                                                {
+                                                    "__typename": "C",
+                                                    "i": {
+                                                        "x": 1,
+                                                        "y": 1
+                                                    },
+                                                    "id": "1"
+                                                }
+                                            ]
+                                        }
+                                    }
+                                },
+                                "response": {
+                                    "body": {
+                                        "data": {
+                                            "_entities": [
+                                                {
+                                                    "c": "c"
+                                                }
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        ]
+                    }
+                }
+            }
+        },
+        {
+            "type": "Request",
+            "request": {
+                "query": "query TestItf2 { req { id c } }"
+            },
+            "expected_response": {
+                "data": {
+                    "req": {
+                        "id": "1",
+                        "c": "c"
+                    }
+                }
+            }
+        },
```

Comment:
this is now failing with the following log:
```
query: {"query":"query TestItf2 { req { id c } }"}

subgraphs received requests:
        method: POST
        path: http://localhost/
        {"query":"query TestItf2__accounts__0{req{__typename id i{__typename id x}}}","operationName":"TestItf2__accounts__0"}

        method: POST
        path: http://localhost/
        {"query":"query TestItf2__products__1($representations:[_Any!]!){_entities(representations:$representations){...on I{y}}}","operationName":"TestItf2__products__1","variables":{"representations":[{"__typename":"I","id":"1"}]}}

        method: POST
        path: http://localhost/
        {"query":"query TestItf2__reviews__2($representations:[_Any!]!){_entities(representations:$representations){...on C{c}}}","operationName":"TestItf2__reviews__2","variables":{"representations":[{"__typename":"C","i":{"x":1,"y":null},"id":"1"}]}}

assertion `left == right` failed
expected: {"data":{"req":{"id":"1","c":"c"}}}
received: {"data":{"req":{"id":"1","c":null}},"errors":[{"message":"HTTP fetch failed from 'reviews': 404: Not Found","path":["req"],"extensions":{"code":"SUBREQUEST_HTTP_ERROR","service":"reviews","reason":"404: Not Found","http":{"status":404}}},{"message":"HTTP fetch failed from 'reviews': subgraph response does not contain 'content-type' header; expected content-type: application/json or content-type: application/graphql-response+json","path":["req"],"extensions":{"code":"SUBREQUEST_HTTP_ERROR","service":"reviews","reason":"subgraph response does not contain 'content-type' header; expected content-type: application/json or content-type: application/graphql-response+json","http":{"status":404}}}]}
```

The request to the reviews subgraph should have the following variables: `"representations":[{"__typename":"C","i":{"x":1,"y":1},"id":"1"}]}` but `y` is null.

this is the content of the fetch node's `requires` field: `[InlineFragment(InlineFragment { type_condition: Some(\"C\"), selections: [Field(Field { alias: None, name: \"__typename\", selections: None }), Field(Field { alias: None, name: \"i\", selections: Some([Field(Field { alias: None, name: \"x\", selections: None }), Field(Field { alias: None, name: \"y\", selections: None })]) }), Field(Field { alias: None, name: \"id\", selections: None })] })]` 
and the available data when trying to extract variables:
`{\"req\":{\"__typename\":\"C\",\"id\":\"1\",\"i\":{\"__typename\":\"A\",\"id\":\"1\",\"x\":1}}}`

Reply:
this is the generated query plan:
```
QueryPlan {
  Sequence {
    Fetch(service: "accounts") {
      {
        req {
          __typename
          id
          i {
            __typename
            id
            x
          }
        }
      }
    },
    Flatten(path: "req.i") {
      Fetch(service: "products") {
        {
          ... on I {
            __typename
            id
          }
        } =>
        {
          ... on I {
            y
          }
        }
      },
    },
    Flatten(path: "req") {
      Fetch(service: "reviews") {
        {
          ... on C {
            __typename
            i {
              x
              y
            }
            id
          }
        } =>
        {
          ... on C {
            c
          }
        }
      },
    },
  },
}
```

Reply:
this was an issue in mocking too